cmake/modules/RootMacros.cmake

```
@@ -1987,38 +1986,38 @@ macro(ROOT_ADD_COMPILE_OPTIONS flags)
 endmacro()
 
 #----------------------------------------------------------------------------
-# find_python_module(module [REQUIRED] [QUIET])
+# ROOT_FIND_PYTHON_MODULE(module [REQUIRED] [QUIET])
+# Set the variable ROOT_PY_<MODULE>_FOUND (all upper case) to the result of
+# the search for the requested Python module and cache the result.
 #----------------------------------------------------------------------------
-function(find_python_module module)
-   CMAKE_PARSE_ARGUMENTS(ARG "REQUIRED;QUIET" "" "" ${ARGN})
-   string(TOUPPER ${module} module_upper)
-   if(NOT PY_${module_upper})
-      if(ARG_REQUIRED)
-         set(py_${module}_FIND_REQUIRED TRUE)
-      endif()
-      if(ARG_QUIET)
-         set(py_${module}_FIND_QUIETLY TRUE)
-      endif()
-      # A module's location is usually a directory, but for binary modules
-      # it's a .so file.
-      execute_process(COMMAND "${Python3_EXECUTABLE}" "-c"
-         "import re, ${module}; print(re.compile('/__init__.py.*').sub('',${module}.__file__))"
-         RESULT_VARIABLE _${module}_status
-         OUTPUT_VARIABLE _${module}_location
-         ERROR_VARIABLE _${module}_error
-         OUTPUT_STRIP_TRAILING_WHITESPACE
-         ERROR_STRIP_TRAILING_WHITESPACE)
-      if(NOT _${module}_status)
-         set(PY_${module_upper} ${_${module}_location} CACHE STRING "Location of Python module ${module}")
-         mark_as_advanced(PY_${module_upper})
+function(ROOT_FIND_PYTHON_MODULE module)
+  CMAKE_PARSE_ARGUMENTS(ARG "REQUIRED;QUIET" "" "" ${ARGN})
+  string(TOUPPER ${module} module_upper)
+  set(FOUND_VAR ROOT_PY_${module_upper}_FOUND)
+
+  if(NOT DEFINED ${FOUND_VAR})
+    execute_process(COMMAND "${Python3_EXECUTABLE}" "-c" "import ${module}"
+      RESULT_VARIABLE status
+      ERROR_QUIET)
+
+    if(${status} EQUAL 0)
+      set(${FOUND_VAR} ON CACHE BOOL "Enable tests depending on '${module}'")
+    else()
+      set(${FOUND_VAR} OFF CACHE BOOL "Enable tests depending on '${module}'")
```

I guess there's a typo?
```suggestion
      set(${FOUND_VAR} OFF CACHE BOOL "Disable tests depending on '${module}'")
```

---

It's not a typo. This comment is for your `CMakeCache`, where it will show as:
```
// Enable tests depending on numba
PY_NUMBA_FOUND:BOOL=OFF
```
So I think it makes sense to keep the comment the same, so users can configure them as ON/OFF if they would e.g. decide to install the python package and enable the tests.

---

But looking at it now, it should probably be called `ROOT_PY_NUMBA_ENABLED:BOOL=OFF`, what do you think?